src/components/pages/admin/events/EventUpdate.js

```diff
@@ -446,6 +453,30 @@ class Event extends Component {
 		);
 	}
 
+	renderLoader() {
+		const { isSubmitting, isSavingDraft, isPublishing } = this.state;
+		const { id } = eventUpdateStore;
+
+		let loadingText = "";
+		if (id) {
+			if (isSavingDraft) {
+				loadingText = "Saving draft...";
+			} else if (isPublishing) {
+				loadingText = "Publishing event...";
+			} else {
+				loadingText = "Updating event...";
+			}
+		} else {
+			loadingText = "Creating new event...";
+		}
+
+		return (
+			<Dialog open={isSubmitting}>
```

Comment:
Will this only show up for submitting but not for publishing and updating?
```suggestion
			<Dialog open={isSubmitting || isSavingDraft || isPublishing}>
```

Reply:
It'll show up for any button click (Save draft or publish). The `isSavingDraft` and `isPublishing` are only used to adjust the loading message

Reply:
👍